python/cudf/cudf/_lib/types.pxd

```diff
@@ -14,3 +14,5 @@ ctypedef int32_t underlying_type_t_type_id
 ctypedef bool underlying_type_t_null_policy
 
 cdef dtype_from_column_view(column_view cv)
+
+cdef libcudf_types.data_type dtype_to_data_type(dtype)
```

Comment:
```suggestion
cdef libcudf_types.data_type dtype_to_data_type(dtype) except *
```

This is my fault as I didn't bring it up eariler:

When a `cdef` function returns a non-Python value, it must be declared `except *` in order to be able to propagate exceptions to its caller.